hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/RecordFinalizerBase.java

```diff
@@ -185,7 +186,10 @@ protected Map<TokenID, List<NftTransfer>> nftChangesFrom(
             // will be explicitly set as 0.0.0
             final var builder = NftTransfer.newBuilder();
             if (modifiedNft != null) {
-                builder.receiverAccountID(modifiedNft.ownerId());
+                // If ownerId is null, we assume that the receiver is the treasury
```

Comment:
Great catch! Thanks @georgi-l95 ! This looks correct , but we also need else case on line 184
when `persistedNft == null` set the `senderAccountId` to  `ZERO_ACCOUNT_ID`

Reply:
Which HAPI test is failing? @georgi-l95

Reply:
@georgi-l95  This test is failing in mono-service also. We should add the change I suggested and update test. I see @mhess-swl  is also working on this ..

Reply:
Yes, I will close this one as Matt already fixed it in his.